### PR TITLE
Add required conditions for Logger's metadata to be print

### PR DESCRIPTION
**Pull request by Krapaince**
Addresses #13772

### CODE REVIEW

File: lib/logger/lib/logger/formatter.ex [hunk 1]
```diff
@@ -74,6 +74,18 @@ defmodule Logger.Formatter do
   to the current process. The user can configure the backend to choose
   which metadata it wants to print and it will replace the `$metadata`
   value.
+
+  > #### Disclaimer {: .warning}
```
Comment (whatyouhide) [inline lib/logger/lib/logger/formatter.ex:78]:
```suggestion
  > #### When is user metadata printed? {: .warning}
```

File: lib/logger/lib/logger/formatter.ex [hunk 2]
```diff
@@ -74,6 +74,18 @@ defmodule Logger.Formatter do
...
+  >
+  > User's metadata must meet one of the following conditions to be printed:
+  >   * Be a string (match with the `is_binary/1` guard)
+  >   * Be a number (either match with the `is_integer/1` or `is_float/1` guard)
```
Comment (whatyouhide) [inline lib/logger/lib/logger/formatter.ex:82]:
```suggestion
  >   * Be a string (`is_binary/1`)
  >   * Be a number (either `is_integer/1` or `is_float/1`)
```

File: lib/logger/lib/logger/formatter.ex [hunk 3]
```diff
@@ -74,6 +74,18 @@ defmodule Logger.Formatter do
...
+  >   * Be a pid
```
Comment (whatyouhide) [inline lib/logger/lib/logger/formatter.ex:83]:
```suggestion
  >   * Be a PID
```

File: lib/logger/lib/logger/formatter.ex [hunk 4]
```diff
@@ -74,6 +74,18 @@ defmodule Logger.Formatter do
...
+  >   * Be an atom
+  >   * Be a reference
+  >   * Be a port
+  >   * Implement the `String.Chars` protocol
+  > otherwise the given metadata will be discarded.
```
Comment (whatyouhide) [inline lib/logger/lib/logger/formatter.ex:88]:
```suggestion
  >
  > If none of the conditions above are true, the given metadata gets discarded.
```

**Review by whatyouhide — APPROVED** [review]

File: lib/logger/lib/logger/formatter.ex [hunk 5]
```diff
@@ -74,6 +74,20 @@ defmodule Logger.Formatter do
   to the current process. The user can configure the backend to choose
   which metadata it wants to print and it will replace the `$metadata`
   value.
+
+  > #### When is user metadata printed? {: .warning}
+  >
+  > User's metadata must meet one of the following conditions to be printed:
```
Comment (josevalim) [inline lib/logger/lib/logger/formatter.ex:80]:
```suggestion
  > User's metadata must meet one of the following conditions to be printed:
  >
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**